rfcs/react-components/non-core-components.md

```diff
@@ -0,0 +1,388 @@
+        - [Pros](#pros-3)
+        - [Cons](#cons-3)
+      - [Separate monorepo](#separate-monorepo)
+        - [Pros](#pros-4)
+        - [Cons](#cons-4)
+    - [Package scope](#package-scope)
+      - [@fluentui scope](#fluentui-scope)
+        - [Pros](#pros-5)
+        - [Cons](#cons-5)
+      - [New package scope](#new-package-scope)
+        - [Pros](#pros-6)
+        - [Cons](#cons-6)
+    - [How users should consume components](#how-users-should-consume-components)
+      - [Suite packages](#suite-packages)
+        - [Pros](#pros-7)
+        - [Cons](#cons-7)
+      - [Individual packages](#individual-packages)
+        - [Pros](#pros-8)
+        - [Cons](#cons-8)
+    - [How to depend on core components](#how-to-depend-on-core-components)
+      - [Depend directly on components packages](#depend-directly-on-components-packages)
+        - [Pros](#pros-9)
+        - [Cons](#cons-9)
+      - [Depend on suite package is peer dependency](#depend-on-suite-package-is-peer-dependency)
+      - [Pros](#pros-10)
+      - [Cons](#cons-10)
+    - [Documentation](#documentation)
+      - [Current docsite](#current-docsite)
+        - [Pros](#pros-11)
+        - [Cons](#cons-11)
+      - [New docsite](#new-docsite)
+        - [Pros](#pros-12)
+        - [Cons](#cons-12)
+    - [Parnter ownership and collaboration](#parnter-ownership-and-collaboration)
+      - [GitHub](#github)
+        - [Pros](#pros-13)
+        - [Cons](#cons-13)
+      - [ADO mirroring](#ado-mirroring)
+        - [Pros](#pros-14)
+        - [Cons](#cons-14)
+      - [Both](#both)
+        - [Pros](#pros-15)
+        - [Cons](#cons-15)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# RFC: Support for components in the Fluent UI umbrella that are not core
+
+[@ling1726](https://github.com/ling1726) [@layershifter](https://github.com/layershifter)
+
+## Summary
+
+@fluentui/react-components has matured, with many flexible primitives that are also highly customizable. These
+primitive components generally don't reuse each other and are considered the 'building blocks' of a design system.
+Future design iterations might introduce components that are either:
+
+- Very situational - i.e. applied to specific scenarios only
+- Composed of multiple primitive components
+- Specializations of primitive components
+
+We need to formalise how we will deliver these components not only as first party packages but also second party
+packages with our partners who might need more situational components powered by V9. This RFC refers to these kinds
+of components as `non-core`.
+
+## Problem statement
+
+We create components that will not be shipped as a part of our 'core' components under the `@fluentui/react-components`
+library but other packages. We will have to finalize the following requirements:
+
+- Should non-core components be able to deviate from V9 core concepts?
+- Should non-core components have a faster breaking change cadence?
```

Comment:
I think yes, they should be versioned independently. They should have a peer dependency on core components.

Reply:
in Fluent AI (copilot) we are building sub 1.0 controls, so we can break going from 0.2.0 to 0.3.0

I think there is still value in moving towards a stable release and using semver properly, but I could see some extended control sitting at 0.215.1 for a while

Reply:
If scenario where fluent extension adhere to fluent core standards, based on recent discussions there should be no breaking changes  ( breaking changes may/will happen after particular major bumps - removal of APIs etc ).

We already started go 1.0 route within core so that should probably be followed. Ideally this will also drive us to establish fluent core which will be used within react-components suite and extensions.

Reply:
Our breaking change policy for core/v9 is highly restrictive, for the reason that primitives that everyone builds on and builds shared experience based on needs to be highly stable. I would not expect any other teams to follow it, and if we don't need to follow it for higher-level experiences we should not restrict ourselves to it (but also remember that it's very important for primitives, or components that other teams build on heavily).

I think the leaf nodes, and non-core components could, and honestly should, have breaking changes more often. They should, of course, follow semver. And whether it's 0.x.y or x.y.z versioning they should follow semver and rev appropriate when they have breaking changes.